### PR TITLE
Feature/update image svg patterns

### PR DESCRIPTION
**Pull request by jryanconklin**
This PR refactors the Image and SVG patterns for a simpler dev experience.

### CODE REVIEW

File: source/default/_patterns/00-protons/index.js [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 import './_base.scss';
 
+// Import Astrogoat from Images.
+import '../../images/astrogoat.png';
```
Comment (illepic) [inline source/default/_patterns/00-protons/index.js]: Noting moving to components.

File: source/default/_patterns/01-atoms/image/__tests__/image.test.js [hunk 2]
```diff
@@ -1,7 +1,5 @@
-import { name } from '..';
+import { name } from '../index';
```
Comment (illepic) [inline source/default/_patterns/01-atoms/image/__tests__/image.test.js]: Just make `..`

File: source/default/_patterns/01-atoms/image/_image--picture.twig [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ *   Implements images with picture properties.
+ *
+ * Available variables for image:
+ *  - Inherit variables from @atoms/_image.twig
+ *  - image.srcset [array]: An array of image src to pull from (falls back to image.src)
+ *      @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture
+ */
+#}
+
+{% set bp = [
```
Comment (illepic) [inline source/default/_patterns/01-atoms/image/_image--picture.twig]: Update via: https://twigfiddle.com/ah3z4a

File: source/default/_patterns/01-atoms/image/_image--primary.twig [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+ * @file
+ * image--primary
+ *   Implements images src and alt.
+ *
+ * Available variables for image:
+ *  - Inherit variables from @atoms/_image.twig
+ */
+#}
+
+<img src="{{ image.src }}" alt="{{ image.alt }}" class="{{ image_classes }}">
```
Comment (illepic) [inline source/default/_patterns/01-atoms/image/_image--primary.twig]: Self closing tag.

File: source/default/_patterns/01-atoms/image/_image.twig [hunk 5]
```diff
@@ -1,43 +1,47 @@
...
+#}
 
-  {# `imageStyles` defined in `_data/imageStyles.yaml` #}
-  {% set imageStyle = imageStyles[imageStyleName] %}
 
-  {% set imageText = imageStyle.width ~ 'x' ~ imageStyle.height %}
+{% set image_classes = image_classes|default([])|merge([
+  'image',
+  'img-fluid',
+  image.style ? 'image--' ~ image.style : '',
```
Comment (illepic) [inline source/default/_patterns/01-atoms/image/_image.twig]: `?:`

File: source/default/_patterns/01-atoms/image/_image.twig [hunk 6]
```diff
@@ -1,43 +1,47 @@
...
-    {% set imageStyle = imageStyle|merge({height: height}) %}
-    {% set imageText = imageStyle.width ~ 'wide, height varies' %}
-  {% endif %}
+{% if image.placeholder %}
+  {% include '@atoms/_image--placeholder.twig' with {
+    image: image,
+    image_classes: image_classes
+  } %}
+
+{% elseif image.variant == 'primary' and not image.placeholder %}
```
Comment (illepic) [inline source/default/_patterns/01-atoms/image/_image.twig]: Can we not `and not` three times?

File: source/default/_patterns/01-atoms/svg/_svg--icon.twig [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+ *    - svgicon.svgname [string]: String filename of the SVG to use from svg/
+ *    - svgicon.svgfill [string]: Pass in a theme-color to colorize the SVG.
+ *    Otherwise it will default to black.
+ */
+#}
+
+{% set svg_classes = svg_classes|default([])|merge([
+  'svg',
+  'svg--icon',
+  svgicon.svgfill ? 'color-fill--' ~ svgicon.svgfill : '',
```
Comment (illepic) [inline source/default/_patterns/01-atoms/svg/_svg--icon.twig]: `?:`

File: source/default/_patterns/01-atoms/svg/fontawesome.js [hunk 8]
```diff
@@ -4,17 +4,18 @@
  * https://fontawesome.com/how-to-use/with-the-api/setup/getting-started
  */
 
-import { library, dom } from '@fortawesome/fontawesome-svg-core';
+import { library, dom } from '@fortawesome/fontawesome-svg-core/index';
```
Comment (illepic) [inline source/default/_patterns/01-atoms/svg/fontawesome.js]: Nuke index

File: source/default/_patterns/01-atoms/svg/fontawesome.js [hunk 9]
```diff
@@ -4,17 +4,18 @@
...
 // Import specific icons required
 import {
   faUserAstronaut,
   faRocket,
   faSpaceShuttle,
   faUser,
-} from '@fortawesome/free-solid-svg-icons';
+  faArrowRight,
```
Comment (illepic) [inline source/default/_patterns/01-atoms/svg/fontawesome.js]: Nuke faArrowRight

File: source/default/_patterns/01-atoms/svg/fontawesome.js [hunk 10]
```diff
@@ -4,17 +4,18 @@
...
+} from '@fortawesome/free-solid-svg-icons/index';
```
Comment (illepic) [inline source/default/_patterns/01-atoms/svg/fontawesome.js]: Nuke index.

File: source/default/_patterns/01-atoms/image/_image--picture.twig [hunk 11]
```diff
@@ -11,16 +11,11 @@
...
-  "md",
-  "lg",
-  "xl"
-] %}
+{% set bp = scssBreakpoints|keys %}
 
 <picture>
   {% for item in image.srcset %}
-    {% set key = (bp[loop.index0]|length) ? bp[loop.index0] : 'xl' %}
+    {% set key = (bp[loop.index]|length) ? bp[loop.index] : 'xl' %}
```
Comment (illepic) [inline source/default/_patterns/01-atoms/image/_image--picture.twig]: Change to:

```twig
 {% set key = bp[loop.index]|default('xl') %}
```

**Comment by adamjuran** [comment]
Hi, I know this is closed but it seemed the best place to ask this: does the image atom now require "variant" to be set in order to output anything at all?

**Comment by jryanconklin** [comment]
> Hi, I know this is closed but it seemed the best place to ask this: does the image atom now require "variant" to be set in order to output anything at all?

Howdy! The short answer is yes, it requires the variant declaration in the pattern.

I've been working this pattern for awhile, but mainly I wanted a declarative approach to images to handle the variances in markup / templates we use with better readability. The old version had a fair bit of magic involved and wasn't a good developer experience.

This version allows:
* Application to Choose Placeholder (grayboxes / holder.js)
* Implement Background Images
* Implement Picture Element

One improvement that could be made is setting a default variant in the pattern, something to the effect of:

```
{% set image_variant = image.variant|default('primary') %}
```

Of course you'd have to reset the if checks that surround the template includes to check `image_variant` rather than `image.variant`.

**Comment by adamjuran** [comment]
Thanks!

I was going to do something like that, but since we're using responsive images with lazysizes integration, which sets "output_image_tag" to true, I just replaced the card_image block with the image output from Drupal. At some point in the future I'll have our devs unset output_image_tag to allow for viewport orientation responsiveness, but as Aragorn once said: "Today is not that day."

**Comment by jryanconklin** [comment]
> Thanks!
> 
> I was going to do something like that, but since we're using responsive images with lazysizes integration, which sets "output_image_tag" to true, I just replaced the card_image block with the image output from Drupal. At some point in the future I'll have our devs unset output_image_tag to allow for viewport orientation responsiveness, but as Aragorn once said: "Today is not that day."

Yeah! I'm definitely using Responsive Image on my project, but I haven't backfilled Particle back with lessons learned quite yet.

**Comment by adamjuran** [comment]
@jryanconklin hope you are doing well (given current circumstances)! I'm now wanting to use bg images in my SASS files and I'm getting webpack errors. Is that type of thing not default in Particle anymore?

**Comment by jryanconklin** [comment]
@adamjuran, doing well enough thanks for asking! You should absolutely still be able to but you may have to reconfigure loaders to handle it since it will require the asset available to file loader.

**Pull request merged**